_source/_posts/2021-09-01-flask-scim-server.md

```diff
@@ -0,0 +1,562 @@
+
+8. Everything should be set up now to run the server locally. Finally, run `python app.py` to do so. You should now have your SCIM server running on `http://localhost:5000`.
+
+### Set Up Ngrok to Route Requests From Okta to Localhost
+
+
+Once you have ngrok installed, run `./ngrok http 5000` to create a tunnel from ngrok to your `http://localhost:5000`. Copy the `https` Forwarding URL created by ngrok as you will need it later.
+
+{% img blog/flask-scim-server/ngrok_url.png alt:"ngrok forwarding URL" width:"800" %}{: .center-image }
+
+### Creating and Configuring Your Okta Application
+
+
+Now it's time to create a new SCIM integration in Okta. If your SCIM app(s) are already set up on the Okta side, feel free to skip ahead to [**Test the SCIM Server**](#test-the-scim-server). There are two options that will work with this server, and I will ALWAYS recommend the first, which is to use an Okta SCIM template application.
+
+#### Option 1: SCIM Template App
+
+1. In your Okta dashboard, go to **Applications** -> **Applications**, then click the **Browse App Catalogue** button. Search for **SCIM 2.0 Test App (Header Auth)** and click the **Add** button once you have it pulled up.
+2. In the **General Settings** tab, click **Next**.
+3. We will set this up as a SWA application, so in the **Sign-On Options** tab, click **Secure Web Authentication**.
+4. Click **Done**.
+5. Tab over to **Provisioning** and click **Configure API Integration.**
+6. Check **Enable API integration**.
+7. In the Base URL field, paste in the ngrok url you generated above with **/scim/v2** appended to the end. In the API Token field, type **Bearer 123456789**. (Later on we will go over how to customize this auth header, but out-of-the-box, the SCIM server expects this value.)
+8. Click **Test API Credentials** and you should get a success message like the below:
+
+    {% img blog/flask-scim-server/scim_1.png alt:"Test Api Credentials UI Image" width:"800" %}{: .center-image }
+
+    > You can check the ngrok logs by navigating to `http://localhost:4040` to see the request from Okta as well as the response from the SCIM server.
+
+    {% img blog/flask-scim-server/scim_2.png alt:"Test Api Credentials Call Image" width:"800" %}{: .center-image }
+
+9. Click **Save**. Now your **Provisoning** tab will look a bit different.
+
+10. Click **Edit** next to **Provisioning to App** and check off:
+    - **Create Users**
+    - **Update User Attributes**
+    - **Deactivate Users**
+    
+And **Save**.
+
+{% img blog/flask-scim-server/scim_3.png alt:"Provisioning Settings Image" width:"800" %}{: .center-image }
+
+#### Option 2: Enable SCIM Provisioning for Existing App Integration Wizard (AIW) App
+
+> Feel free to skip over this section to **Test the SCIM Server** if you set your SCIM integration up above.
```

Comment:
```suggestion
> Feel free to skip over this section to **Test the SCIM Server** if you set up your SCIM integration above.
```
instead of splitting "set up"